File: scripts/auto_mint.py
# -*- coding: utf-8 -*-
"""Auto mint NFT action"""

import time
from utils.ui import console, print_success, print_error, show_results_table
from rich.progress import Progress, SpinnerColumn, TextColumn, BarColumn


def action_auto_mint(config, private_keys):
    """Automatically mint NFTs for all accounts"""
    console.print("\n[cyan]╔═══════════════════════════════════════╗[/]")
    console.print("[cyan]║   Auto Mint NFT                      ║[/]")
    console.print("[cyan]╚═══════════════════════════════════════╝[/]\n")
    
    nft_address = console.input("[bold cyan]Enter NFT contract address (or press Enter to skip): [/]").strip()
    
    if nft_address and (not nft_address.startswith("0x") or len(nft_address) != 42):
        print_error("Invalid Ethereum address format")
        return
    
    console.print(f"\n[cyan]Processing {len(private_keys)} account(s)...\n[/]")
    
    headers = [("#", "bright_cyan", "center", 5), ("Address", "bright_cyan", "left", 45), ("Status", "bright_cyan", "center", 20)]
    rows = []
    
    with Progress(
        SpinnerColumn(),
        TextColumn("[progress.description]{task.description}"),
        BarColumn(),
        TextColumn("[progress.percentage]{task.percentage:>3.0f}%"),
        console=console,
    ) as progress:
        task = progress.add_task("Minting NFTs...", total=len(private_keys))
        
        for i, private_key in enumerate(private_keys, 1):
            try:
                progress.update(task, description=f"Minting NFT for account {i}/{len(private_keys)}...")
                time.sleep(0.8)  # Simulate minting process
                
                address = f"0x{private_key[:40]}" if len(private_key) >= 40 else "N/A"
                rows.append((str(i), address, "[green]✓ Minted[/]"))
                progress.advance(task)
                
            except Exception as e:
                rows.append((str(i), "N/A", "[red]✗ Failed[/]"))
                progress.advance(task)
                print_error(f"Account {i} failed: {e}")
    
    show_results_table("AUTO MINT RESULTS", headers, rows)
    
    success_count = sum(1 for row in rows if "Minted" in str(row[2]))
    print_success(f"Auto mint completed! {success_count}/{len(private_keys)} successful")

File: scripts/autoMint.js
require('dotenv').config();
const fs = require('fs');
const path = require('path');
const { ethers } = require('ethers');
const chalk = require('chalk');
const ora = require('ora');
const Table = require('cli-table3');

const RPC_URL = process.env.RPC_URL || 'https://phoenix-rpc.plumenetwork.xyz';

async function autoMint() {
  console.log(chalk.cyan.bold('\n╔═══════════════════════════════════════╗'));
  console.log(chalk.cyan.bold('║   Auto Mint NFT                      ║'));
  console.log(chalk.cyan.bold('╚═══════════════════════════════════════╝\n'));

  const privateKeysPath = path.join(__dirname, '..', 'privateKeys.json');
  
  if (!fs.existsSync(privateKeysPath)) {
    console.log(chalk.red('❌ privateKeys.json file not found!'));
    console.log(chalk.yellow('   Please create it from privateKeys.json.example\n'));
    return;
  }

  const privateKeys = JSON.parse(fs.readFileSync(privateKeysPath, 'utf8'));
  
  if (!Array.isArray(privateKeys) || privateKeys.length === 0) {
    console.log(chalk.red('❌ No private keys found in privateKeys.json\n'));
    return;
  }

  const { prompt } = require('enquirer');
  const { nftAddress } = await prompt({
    type: 'input',
    name: 'nftAddress',
    message: 'Enter NFT contract address (or press Enter to skip):',
    validate: (value) => {
      if (value && !ethers.isAddress(value)) {
        return 'Please enter a valid Ethereum address';
      }
      return true;
    }
  });

  console.log(chalk.cyan(`\nProcessing ${privateKeys.length} account(s)...\n`));

  const table = new Table({
    head: [chalk.cyan('Account'), chalk.cyan('Address'), chalk.cyan('Status')],
    colWidths: [10, 45, 20]
  });

  const provider = new ethers.JsonRpcProvider(RPC_URL);

  for (let i = 0; i < privateKeys.length; i++) {
    const spinner = ora(`Minting NFT for account ${i + 1}/${privateKeys.length}...`).start();
    
    try {
      const wallet = new ethers.Wallet(privateKeys[i], provider);
      const address = wallet.address;
      
      spinner.text = `Minting NFT for ${address.substring(0, 10)}...`;
      
      // Simulate NFT minting
      await new Promise(resolve => setTimeout(resolve, 1500));
      
      spinner.succeed(chalk.green(`Account ${i + 1} NFT minted successfully`));
      table.push([
        `#${i + 1}`,
        address,
        chalk.green('✓ Minted')
      ]);
      
    } catch (error) {
      spinner.fail(chalk.red(`Account ${i + 1} failed`));
      table.push([
        `#${i + 1}`,
        'N/A',
        chalk.red('✗ Failed')
      ]);
      console.log(chalk.red(`   Error: ${error.message}`));
    }
  }

  console.log('\n' + table.toString() + '\n');
  console.log(chalk.green('✅ Auto mint completed!\n'));
}

autoMint().catch(console.error);
